ldk/javascript/examples/self-test-loop/src/tests/whisper.ts

```diff
@@ -0,0 +1,700 @@
+|Back Pain|Earache|
+|F/U Hypertension/Blood Pressure|TB Test|
+|||
+|||
+|||
+|||
+|**OB Video Visit**||
+|Contraceptive Consults|F/U Labs/Tests/Ultrasounds|
+|Infertility Consults|Post-Partum Appointments (Scheduled By Office)|
+            `,
+          },
+          {
+            type: whisper.WhisperComponentType.Markdown,
+            body: `
+# Box in the Box Example
+`,
+          },
+          {
+            type: whisper.WhisperComponentType.Box,
+            alignment: Alignment.Center,
+            direction: Direction.Horizontal,
+            children: [
+              {
+                type: whisper.WhisperComponentType.Box,
+                alignment: Alignment.Left,
+                direction: Direction.Vertical,
+                children: [
+                  {
+                    type: whisper.WhisperComponentType.Markdown,
+                    body: `
+**Header Left**
+
+Some text on the left
+`,
+                  },
+                  {
+                    type: whisper.WhisperComponentType.TextInput,
+                    label: 'Left Input',
+                    onChange: (value) => {
+                      console.debug(`Input value changed: ${value}`);
+                    },
+                  },
+                ],
+              },
+              {
+                type: whisper.WhisperComponentType.Box,
+                alignment: Alignment.Right,
+                direction: Direction.Vertical,
+                children: [
+                  {
+                    type: whisper.WhisperComponentType.Markdown,
+                    body: `
+**Header Right**
+
+Some text on the right
+`,
+                  },
+                  {
+                    type: whisper.WhisperComponentType.TextInput,
+                    label: 'Right Input',
+                    onChange: (value) => {
+                      console.debug(`Input value changed: ${value}`);
+                    },
+                  },
+                ],
+              },
+            ],
+          },
+          {
+            type: whisper.WhisperComponentType.Box,
+            alignment: Alignment.Left,
+            direction: Direction.Horizontal,
+            children: [
+              {
+                type: whisper.WhisperComponentType.Button,
+                buttonStyle: ButtonStyle.Primary,
+                label: 'Press if Rendered',
+                onClick: () => {
+                  form.close((error) => console.log(error));
+                  resolve(true);
+                },
+              },
+              {
+                type: whisper.WhisperComponentType.Button,
+                buttonStyle: ButtonStyle.Secondary,
+                label: 'Press if NOT Rendered',
+                onClick: () => {
+                  form.close((error) => console.log(error));
+                  reject(false);
+                },
+              },
+            ],
+          },
+        ],
+      });
+    } catch (error) {
+      console.error(error);
```

Comment:
Might want to close the whispers as well upon rejection.

Reply:
Even though I don't believe you added this.

Reply:
I added a timeout form close to close the whisper if no action from the user received. Although we cant do it in the catch as it means that whisper is not properly created and there is no instance to close.